test/ng/compileSpec.js

```diff
@@ -2762,6 +2762,23 @@ describe('$compile', function() {
     }));
 
 
+    it('should handle consecutive text elements as a single text element', inject(function($rootScope, $compile) {
+      window.addMutationObserver = function() {
```

Comment:
If I recall correctly, the issue appeared when there was a childList+subtree MutationObserver **registered** on the page. I would expect a test that calls `new MutationObserver(...).observe(...)`.

Instead I see:

```js
window.addMutationObserver = function () {
  // A function that _would_ register a MutationObserver on body
  // IF it where executed.
};

// Then I see the actual test code being executed without ever calling `window.addMutationObserver()`
// which would have registered the MutationObserver.
```

So, afaict, there is **no** MutationObserver during this test, so it would pass even without the fix (I am just speculating - I haven't actually tried it).

--
I would expect something along these lines:

```js
it ('...', inject(function($compile, $rootScope) {
  // No point it running thetest, if there is no MutationObserver
  if (!window.MutationObserver) return;

  // Create and register the MutationObserver
  var observer = new window.MutationObserver(noop);
  observer.observe(document.body, {childList: true, sublist: true});

  // Run the actual test
  var base = jqLite('<div>&mdash; {{ "This doesn\'t." }}</div>');
  element = $compile(base)($rootScope);
  $rootScope.$digest();
  expect(element.text()).toBe("— This doesn't.");

  // Unregister the MutationObserver (and hopeit doesn't mess up with subsequent tests)
  observer.disconnect();
}));
```